python/tskit/trees.py

```diff
@@ -833,6 +833,37 @@ def unrank(rank, num_leaves):
         """
         return combinatorics.RankTree.unrank(rank, num_leaves).to_tsk_tree()
 
+    def count_topologies(self, key=None):
+        """
+        Computes the distribution of leaf-labelled tree topologies
+        labelled by ``key`` that are exhibited by the samples of ``tree``.
+        ``key`` defaults to selecting the population of each sample node,
+        therefore the result would be the distribution of evolutionary topologies
+        of the populations represented by the samples in ``tree``.
+        Internal samples are not supported.
+
+        Produces a dictionary from a ``frozenset`` of labels to
+        ``collections.Counter`` of topologies, where each tree topology is represnted
+        by its rank (see :ref:`sec_combinatorics`). This represents the
+        topologies for a particular set of labels and the number of times
+        each topology appears when subsampling samples from ``tree``.
+
+        :param tree tskit.Tree: The tree to compute the topologies for.
+        :param key: A function of the form ``key(u)`` where ``u`` is a
```

Comment:
Let's change ``key`` to ``sample_sets`` so that we're using the same concepts as the stats API.